sdk/trace/span.go

```diff
@@ -364,99 +362,90 @@ func (s *recordingSpan) SetName(name string) {
 
 // Name returns the name of this span.
 func (s *recordingSpan) Name() string {
-	s.mu.Lock()
-	defer s.mu.Unlock()
+	s.mu.RLock()
+	defer s.mu.RUnlock()
 	return s.name
 }
 
 // Name returns the SpanContext of this span's parent span.
 func (s *recordingSpan) Parent() trace.SpanContext {
-	s.mu.Lock()
-	defer s.mu.Unlock()
+	s.mu.RLock()
+	defer s.mu.RUnlock()
 	return s.parent
 }
 
 // SpanKind returns the SpanKind of this span.
 func (s *recordingSpan) SpanKind() trace.SpanKind {
-	s.mu.Lock()
-	defer s.mu.Unlock()
+	s.mu.RLock()
+	defer s.mu.RUnlock()
 	return s.spanKind
 }
 
 // StartTime returns the time this span started.
 func (s *recordingSpan) StartTime() time.Time {
-	s.mu.Lock()
-	defer s.mu.Unlock()
+	s.mu.RLock()
+	defer s.mu.RUnlock()
 	return s.startTime
 }
 
 // EndTime returns the time this span ended. For spans that have not yet
 // ended, the returned value will be the zero value of time.Time.
 func (s *recordingSpan) EndTime() time.Time {
-	s.mu.Lock()
-	defer s.mu.Unlock()
+	s.mu.RLock()
+	defer s.mu.RUnlock()
 	return s.endTime
 }
 
 // Attributes returns the attributes of this span.
 func (s *recordingSpan) Attributes() []attribute.KeyValue {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	if s.attributes.evictList.Len() == 0 {
-		return []attribute.KeyValue{}
-	}
-	return s.attributes.toKeyValue()
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	return s.attributes.ToKeyValue()
```

Comment:
This method now needs to acquire two locks instead of one. This does not seem idea.

Reply:
One of them being read-only, it means it can be acquired even if another read-only lock has already been acquired. Whereas with the current implementation, nobody can do anything if anybody is doing something else at the same time.

Since the attributes map within the span is never changed after initialization, we could also remove this lock entirely though.